app/Http/Controllers/Admin/Device/DeviceController.php

```
@@ -14,50 +14,80 @@ class DeviceController extends Controller
+
+    public function destroy(Device $device) {
+
+        $device->delete();
+        return redirect()
+            ->route('devices.index')
+            ->with('success', 'Device trashed.');
+    }
 
-        $device->table_id = $request->table_id;
-        $device->save();
+    public function restore(Request $request, int $id){
```

PHPMD (2.15.0)

Avoid unused parameters such as '$request'. (undefined)

(UnusedFormalParameter)